pkg/storage/export/virt-exportserver/exportserver.go

```diff
@@ -492,6 +497,34 @@ func archiveHandler(mountPoint string) http.Handler {
 	})
 }
 
+func checkDirectoryPermissions(filePath string) bool {
+	dir, err := os.Open(filePath)
+	if err != nil {
+		log.Log.Reason(err).Errorf("error opening %s", filePath)
+		return false
+	}
+	defer dir.Close()
+
+	// Read all filenames
+	contents, err := dir.Readdirnames(-1)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to read directory contents: %v", err)
+		return false
+	}
+
+	for _, item := range contents {
+		itemPath := filepath.Join(filePath, item)
+		// Check if export server has permissions to manipulate the file
+		file, err := os.Open(itemPath)
+		if err != nil {
+			log.Log.Reason(err).Errorf("unable to open %s, file may lack read permissions", itemPath)
```

Comment:
This error message will stutter, os.Open already prints the filename and the reason for failing to open:
```
Error opening file: open test.txt: permission denied
```

Reply:
Ok, I think I can improve this one